demos/sample_azure_iot_pnp/sample_azure_iot_pnp.c

```diff
@@ -0,0 +1,1062 @@
+                {
+                    LogError( ( "Error sending command response" ) );
+                }
+            }
+        }
+        else
+        {
+            LogError( ( "Error generating command payload" ) );
+
+            if( AzureIoTHubClient_SendCommandResponse( xHandle, pxMessage, 501,
+                                                       sampleazureiotCOMMAND_EMPTY_PAYLOAD,
+                                                       ulCommandPayloadLength ) != eAzureIoTSuccess )
+            {
+                LogError( ( "Error sending command response" ) );
+            }
+        }
+    }
+    else
+    {
+        /* Not for max min report (not for this device) */
+        LogInfo( ( "Received command is not for this device" ) );
+
+        if( AzureIoTHubClient_SendCommandResponse( xHandle, pxMessage, 404,
+                                                   sampleazureiotCOMMAND_EMPTY_PAYLOAD,
+                                                   strlen( sampleazureiotCOMMAND_EMPTY_PAYLOAD ) ) != eAzureIoTSuccess )
+        {
+            LogError( ( "Error sending command response" ) );
+        }
+    }
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Properties callback handler
+ */
+static AzureIoTResult_t prvProcessProperties( AzureIoTHubClientPropertiesResponse_t * pxMessage,
+                                              AzureIoTHubClientPropertyType_t xPropertyType,
+                                              double * xOutTemperature,
+                                              uint32_t * ulOutVersion )
+{
+    AzureIoTResult_t xResult;
+    AzureIoTJSONReader_t xReader;
+    AzureIoTJSONTokenType_t xTokenType;
+    const uint8_t * pucComponentName = NULL;
+    uint32_t ulComponentNameLength = 0;
+
+    *xOutTemperature = 0.0;
+
+    xResult = AzureIoTJSONReader_Init( &xReader, pxMessage->pvMessagePayload, pxMessage->ulPayloadLength );
+    configASSERT( xResult == eAzureIoTSuccess );
+
+    xResult = AzureIoTHubClientProperties_GetPropertiesVersion( &xAzureIoTHubClient, &xReader, pxMessage->xMessageType, ulOutVersion );
+
+    if( xResult != eAzureIoTSuccess )
+    {
+        LogError( ( "Error getting the property version" ) );
+    }
+    else
+    {
+        /* Reset JSON reader to the beginning */
+        xResult = AzureIoTJSONReader_Init( &xReader, pxMessage->pvMessagePayload, pxMessage->ulPayloadLength );
+        configASSERT( xResult == eAzureIoTSuccess );
+
+        while( ( xResult = AzureIoTHubClientProperties_GetNextComponentProperty( &xAzureIoTHubClient, &xReader,
+                                                                                 pxMessage->xMessageType, xPropertyType,
+                                                                                 &pucComponentName, &ulComponentNameLength ) ) == eAzureIoTSuccess )
+        {
+            xResult = AzureIoTJSONReader_TokenType( &xReader, &xTokenType );
```

Comment:
>xResult = AzureIoTJSONReader_TokenType( &xReader, &xTokenType ); [](http://example.com/codeflow?start=12&length=64)

why do you need to know this token type, as next line irrespective of token assume token to be string token.